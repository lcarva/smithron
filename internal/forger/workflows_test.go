package forger

import (
	"context"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestWorkflows(t *testing.T) {
	p := Plan{
		Name:    "hello-world",
		Image:   "docker.io/busybox:latest",
		Command: []string{"echo", "Hello, World!"},
		Env: map[string]string{
			"GREETING": "Hello",
		},
	}

	ctx := context.Background()

	tf, err := GetForgerForTarget("argo-workflows")
	require.NoError(t, err)
	require.IsType(t, ArgoWorkflowsForger{}, tf)

	got, err := tf.Forge(ctx, p)
	require.NoError(t, err)
	snaps.MatchYAML(t, got)
}
