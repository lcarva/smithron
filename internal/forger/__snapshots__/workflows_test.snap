
[TestWorkflows - 1]
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  creationTimestamp: null
  name: hello-world
spec:
  arguments: {}
  templates:
  - container:
      command:
      - echo
      - Hello, World!
      env:
      - name: GREETING
        value: Hello
      image: docker.io/busybox:latest
      name: ""
      resources: {}
    inputs: {}
    metadata: {}
    name: run
    outputs: {}

---
